src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationOptions.cs

```diff
@@ -166,7 +172,36 @@ public AzureAppConfigurationOptions UseFeatureFlags(Action<FeatureFlagOptions> c
             {
                 throw new InvalidOperationException($"Please select feature flags by either the {nameof(options.Select)} method or by setting the {nameof(options.Label)} property, not both.");
             }
+
+            // Read Feature Management schema version from environment variables
+            string featureManagementSchemaVersion = null;
+
+            try
+            {
+                featureManagementSchemaVersion = Environment.GetEnvironmentVariable(FeatureManagementConstants.FeatureManagementSchemaEnvironmentVariable);
+            }
+            catch (SecurityException) { }
             
+            if (!string.IsNullOrWhiteSpace(featureManagementSchemaVersion) 
+                && featureManagementSchemaVersion == FeatureManagementConstants.FeatureManagementSchemaV2)
+            {
+                FeatureManagementSchemaVersion = FeatureManagementConstants.FeatureManagementSchemaV2;
```

Comment:
I'd recommend not setting the `FeatureManagementSchemaVersion` member until the adapter is added. The adapter has the if clause to protect against repeated calls. In fact, the environment variable isn't needed unless we are adding the adapter it looks like.